aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -241,6 +246,52 @@ private List<String> deleteObjectsInBucket(String bucket, Collection<String> obj
     return Lists.newArrayList();
   }
 
+  @Override
+  public Iterator<FileInfo> listPrefix(String prefix) {
+    S3URI s3uri = new S3URI(prefix, awsProperties.s3BucketToAccessPointMapping());
+
+    return internalListPrefix(s3uri.bucket(), s3uri.key()).stream()
+        .flatMap(r -> r.contents().stream())
+        .map(o -> new FileInfo(o.key(), o.size(), o.lastModified().toEpochMilli())).iterator();
+  }
+
+  /**
+   * This method provides a "best-effort" to delete all objects under the
+   * given prefix.
+   *
+   * Bulk delete operations are used and no reattempt is made for deletes if
+   * they fail, but will log any individual objects that are not deleted as part
+   * of the bulk operation.
+   *
+   * @param prefix prefix to delete
+   */
+  @Override
+  public void deletePrefix(String prefix) {
```

Comment:
After looking at the bulk delete code, I think we may want to delegate to it, or at least refactor to share.

Deletes now support tag-based deletion, so this needs to detect whether tags should be used. Also, S3 supports bulk deletes, which would make this go much more quickly.

Reply:
I'll take a look at consolidating the two delete paths.  It's a little messy because the bulk delete is not streaming and will pull everything into memory, so doesn't work well for large operations.

I'll see if I can find a combination / compromise here.

Reply:
The bulk delete accepts an Iterable, so I think it should work with streaming.